### ISSUE TITLE
Whatsnew bad redirection when changing version

### ISSUE THREAD

**Issue by ThisisYoYoDev** [python/docsbuild-scripts#138]

**Comment by JulienPalard** [comment]
> FYI @JulienPalard , you can simply transfer this issue directly to that repo

Wow thanks!

**Comment by JulienPalard** [comment]
I think we all agree that something better can be implemented.

I'm not sure we need to complicate things more than just navigating to `f"{selected_version}/whatsnew/{selected_version}.html"`, after all:

- It's simple and predictible.
- The page always exists.
- If the reader choose a version from the picker while reading a whatsnew, it's highly probable that he wanted to see the choosen version (am I wrong?).

**Comment by CAM-Gerlach** [comment]
I certainly understand the desire to not complicate the logic, which was my one concern with this proposal to begin with.

That's essentially what the above proposed implementation boils down to (in perhaps a dozen or so total lines of JS for the whole thing), except that if the reader is browsing a What's New that isn't the current version's, and exists on the target, the What's New version isn't touched, as it wouldn't make sense to arbitrarily change it (given the reader deliberately navigated to that older version, which I would view as a regression).

**Comment by ThisisYoYoDev** [comment]
I completely agree with @JulienPalard what I wanted here and what I think everyone wants is to be redirected to the version they want.

When I arrived on the Whatsnew of 3.11 and I saw that the version 3.12 was available in alpha I said to myself I want to see the news about 3.12

So I think the best thing would be to keep it simple like Julien suggested in his message :)

**Comment by hugovk** [comment]
See also https://github.com/python/cpython/issues/100734.
